### ISSUE TITLE
Compiler should report use of undefined variable when combining exception filtering and pattern matching

### ISSUE THREAD

**Issue by Insomniak47** [dotnet/roslyn#51454]
Hey all, 

Not sure if this falls under the compiler team or the language team. This is something that came up while handling errors in a microservice I'm writing. I tend to heavily use the pattern matching features of the language and so I naturally used the 'is' construct with assignment to pattern match an InnerException in an Exception Filter expression. I didn't know if it would work but it compiled and I think the behaviour was undesirable and probably should be identified

**Version Used**: 
Unsure how to find this but currently using VS2019 Version 16.8.5
**Steps to Reproduce**:

1. Create a catch block that catches using a filter expression on the InnerException of an aggregate exception 

```
catch (DbUpdateException ex ) when (ex.InnerException is PostgresException pg) 
```

2. Attempt to use the members of the pattern matched exception (pg) in the catch block associated with the expression

**Expected Behavior**:

The previous value pg should either:
- Output a compiler warning/error that the value isn't well defined
- Have the value of the member ex.InnerException (object.ReferenceEquals(ex.InnerException,pg) == true)

**Actual Behavior**:
An instance of PostgresException is created with default values passed in as the parameters to its constructor and code that subsequently uses it executes normally except with garbage values.

**Comment by Insomniak47** [comment]
Sorry, going to close this issue. It seems like I was using the wrong param on the exception that I was grabbing but the debugger doesn't think that value is defined at that point so I'll push it off to the debugger team!

Sorry for the trouble
